homeassistant/components/bitvavo/sensor.py

```diff
@@ -0,0 +1,268 @@
+"""Sensor for the Bitvavo integration."""
+import logging
+
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from . import BitvavoDataUpdateCoordinator
+from .const import (
+    ASSET_VALUE_BASE,
+    ASSET_VALUE_CURRENCIES,
+    ATTRIBUTION,
+    CONF_ASSET_TICKERS,
+    CONF_BALANCES,
+    CONF_OPEN_ORDERS,
+    CONF_TICKERS,
+    CURRENCY_ICONS,
+    DEFAULT_COIN_ICON,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Bitvavo sensors."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        Ticker(coordinator, market) for market in coordinator.data[CONF_TICKERS]
+    ]
+
+    if CONF_BALANCES in coordinator.data:
+        for balance in coordinator.data[CONF_BALANCES]:
+            entities.append(Balance(coordinator, balance))
+
+        for currency in ASSET_VALUE_CURRENCIES:
+            entities.append(TotalAssetValue(coordinator, currency))
+
+    if CONF_OPEN_ORDERS in coordinator.data:
+        entities.append(OpenOrders(coordinator, coordinator.data[CONF_OPEN_ORDERS]))
+
+    async_add_entities(entities)
+
+
+class Ticker(CoordinatorEntity):
+    """Implementation of the ticker sensor."""
+
+    def __init__(self, coordinator: BitvavoDataUpdateCoordinator, symbol):
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._symbol = symbol
+
+        self._name = f"Bitvavo Ticker - {self._symbol}"
```

Comment:
You can save a lot of lines by using `_attr_*` and/or `SensorEntityDescription`

Reply:
I believe this is the third time I have to rebase this PR and rewrite/implement new HA features, which where added in missed versions, because PR reviews take so long... it's quite cumbersome, but will look at it.. thanks.